doc/sources/non-scikit-algorithms.rst

```diff
@@ -0,0 +1,157 @@
+.. ******************************************************************************
+.. * Copyright 2024 Intel Corporation
+.. *
+.. * Licensed under the Apache License, Version 2.0 (the "License");
+.. * you may not use this file except in compliance with the License.
+.. * You may obtain a copy of the License at
+.. *
+.. *     http://www.apache.org/licenses/LICENSE-2.0
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+Non-Scikit Algorithms
+=====================
+Algorithms not presented in the original Scikit-learn are described here. All algorithms are 
+available for both CPU and GPU (including distributed mode)
+
+BasicStatistics
```

Comment:
This could be auto-generated from the docstrings of the class using the "napoleon" plugin for Sphinx.

See for example this file in XGBoost:
https://github.com/dmlc/xgboost/blob/master/doc/python/python_api.rst?plain=1

So the process would be as follows:
* Add the napoleon plugin to the configuration. See here for an example: https://github.com/dmlc/xgboost/blob/59e6c924254e29b7b521252af056bd970435b77a/doc/conf.py#L151
* Write all the docstrings for these classes (methods, parameters, attributes, examples, etc.) using NumPy format: https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes
* Add some lines like these, with `autoclass`: https://github.com/dmlc/xgboost/blob/59e6c924254e29b7b521252af056bd970435b77a/doc/python/python_api.rst?plain=1#L23

For autoclass to work, you might need to modify the configuration file to be able to import the modules if they aren't already added:
https://github.com/intel/scikit-learn-intelex/blob/8883b3948eb14a8f69a9d7a741e56d14d2d2df4d/doc/sources/conf.py#L35

Reply:
done